### ISSUE TITLE
Allow Wrapper to handle keep alives

### ISSUE THREAD

**Issue by JasonBristol** [suresttexas00/minecraft-wrapper#5]
What is the intention of the following conditions in proxy.py:

```python
if True == False:  # self.pktCB.blockchange: # Block Change - disabled - not doing anything at this point
            if self.version < PROTOCOLv1_8START:
                # Temporary! These packets need to be filtered for cross-server
                # stuff.
                return True
            data = self.read("position:location|varint:id")
```

```python
if False == True:  # time.time() - self.tPing > 1 and self.state == 3:
                    if self.version > 32:
                        self.send(self.pktCB.keepalive, "varint",
                                  (random.randrange(0, 99999),))
                        if self.clientSettings and not self.clientSettingsSent:
                            # print "Sending self.clientSettings..."
                            # print self.clientSettings
                            if self.version < PROTOCOL_1_9START:
                                self.server.send(self.pktSB.clientsettings, "string|byte|byte|bool|ubyte", (
                                    self.clientSettings["locale"],
                                    self.clientSettings["view_distance"],
                                    self.clientSettings["chat_mode"],
                                    self.clientSettings["chat_colors"],
                                    self.clientSettings["displayed_skin_parts"]
                                ))
                                self.clientSettingsSent = True
                            else:
                                self.server.send(self.pktSB.clientsettings, "string|byte|varint|bool|ubyte|varint", (
                                    self.clientSettings["locale"],
                                    self.clientSettings["view_distance"],
                                    self.clientSettings["chat_mode"],
                                    self.clientSettings["chat_colors"],
                                    self.clientSettings[
                                        "displayed_skin_parts"],
                                    self.clientSettings["main_hand"]
                                ))
                                self.clientSettingsSent = True

                    else:
                        # _OLD_ MC version
                        self.send(0x00, "int", (random.randrange(0, 99999),))
                    self.tPing = time.time()
                if self.parse(id) and self.server:
                    if self.server.state == 3:
                        self.server.sendRaw(original)
```

```python
# handle keep alive packets
        if False == True:  # id == self.pktCB.keepalive and self.state == 3:
            if self.client.version > 7:
                id = self.read("varint:i")["i"]
                if id is not None:
                    self.send(self.pktSB.keepalive, "varint", (id,))
            return False
```

**Comment by suresttexas00** [comment]
Those are disabled items that could not be made to function properly.  The original if statements for each one are behind the '#'.. ie `# id == self.pktCB.keepalive and self.state == 3:`

We should try to get these working.  I'm not sure if letting wrapper handle the keep alives is a good idea or not.. I think 'not', but Ben put those in to allow a future case where a client could be connected to wrapper, but not to a server.

**Comment by JasonBristol** [comment]
Hmm I see now, I'll keep this floating here while this code is inert until we make a determination, going to work on other cleanup items.

**Comment by JasonBristol** [comment]
Wrapper to Wrapper connections would be a really nice feature, this would definitely require Wrapper to handle keep alives, we should consider this work for FP1

**Comment by suresttexas00** [comment]
maybe this is why I have been unable to get player.connect to work now.

**Comment by suresttexas00** [comment]
closed - #120 fixes keep alives, which are now handled on both client and server connection ends.
